receiver/osqueryreceiver/osquery.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package osqueryreceiver
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	osquery "github.com/osquery/osquery-go"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver"
+	semconv "go.opentelemetry.io/collector/semconv/v1.21.0"
+	"go.uber.org/zap"
+)
+
+type osQueryReceiver struct {
+	config       *Config
+	logger       *zap.Logger
+	client       *osquery.ExtensionManagerClient
+	logsConsumer consumer.Logs
+	stopperChan  chan struct{}
+	wg           sync.WaitGroup
+	start        time.Time
+	end          time.Time
+}
+
+func newOsQueryReceiver(cfg *Config, consumer consumer.Logs, set receiver.CreateSettings) *osQueryReceiver {
+	osqr := &osQueryReceiver{
+		config:       cfg,
+		logsConsumer: consumer,
+		stopperChan:  make(chan struct{}),
+		logger:       set.Logger,
+		client:       cfg.getOsQueryClient(),
+	}
+
+	return osqr
+}
+
+func newLog(ld plog.Logs, query string, row map[string]string) plog.Logs {
+	rl := ld.ResourceLogs().AppendEmpty()
+	resourceAttrs := rl.Resource().Attributes()
+	for k, v := range row {
+		resourceAttrs.PutStr(k, v)
+	}
+
+	sl := rl.ScopeLogs().AppendEmpty()
+	sl.Scope().Attributes().PutStr(semconv.AttributeOTelScopeName, "otelcol/osqueryreceiver")
+
+	lr := sl.LogRecords().AppendEmpty()
+	lr.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+	lr.SetSeverityNumber(plog.SeverityNumberInfo)
+	lr.SetSeverityText("INFO")
+	lr.Body().SetStr(query)
+	return ld
+}
+
+func (or *osQueryReceiver) runQuery(ctx context.Context, query string) {
+	if or.client == nil {
+		return
+	}
+
+	rows, err := or.client.QueryRows(query)
```

Comment:
it can happen that client running for some time can run into error similar to `osquery failed: write unix ->/var/run/642722736/osquery.sock: write: broken pipe` 
it would be better to handle this in a graceful way and do reconnect upfront.

Reply:
Good idea, will implement [something similar to what the osquery beats connector is doing](https://github.com/aleksmaus/beats/blob/8982110e828f512db0985ec5737881824b9309ba/x-pack/osquerybeat/internal/osqdcli/client.go#L130) with retries.